sdks/universal-router-sdk/src/swapRouter.ts

```diff
@@ -82,14 +82,25 @@ export abstract class SwapRouter {
     positionManagerOverride?: string
   ): MethodParameters {
     const v4Pool: V4Pool = options.outputPosition.pool
-    const token0 = options.inputPosition.pool.token0
-    const token1 = options.inputPosition.pool.token1
+    const v3Token0 = options.inputPosition.pool.token0
+    const v3Token1 = options.inputPosition.pool.token1
     const v4PositionManagerAddress =
       positionManagerOverride ?? CHAIN_TO_ADDRESSES_MAP[v4Pool.chainId as SupportedChainsType].v4PositionManagerAddress
 
+    const nativeCurrency = Ether.onChain(v4Pool.chainId)
+
     // validate the parameters
-    invariant(token0 === v4Pool.token0, 'TOKEN0_MISMATCH')
-    invariant(token1 === v4Pool.token1, 'TOKEN1_MISMATCH')
+    if (nativeCurrency.equals(v4Pool.currency0)) {
```

Comment:
is nativeCurrency needed for this? could it just be `v4Pool.currency0.isNative`?

Reply:
then you could remove the line above and the import too